### PR TITLE
Converted all the types mentions to the links on the mentioned types pages

### LINKED ISSUE #274: Convert all the types mentions to the links on the mentioned types pages
- [x] GroupBy
- [x] DataFrameSchema

### PR DESCRIPTION
**Pull request by zaleslaw**
Closed #274

### CODE REVIEW

File: docs/StardustDocs/topics/extensionPropertiesApi.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ df.add("lastName") { name.split(",").last() }
 
 <!---END-->
 
-Extension properties are generated for DataSchema that is extracted from [`DataFrame`](DataFrame.md) instance after REPL line execution. After that `DataFrame` variable is typed with its own `DataSchema`, so only valid extension properties corresponding to actual columns in DataFrame will be allowed by the compiler and suggested by completion.
+Extension properties are generated for [`DataSchema`](schemas.md) that is extracted from [`DataFrame`](DataFrame.md) instance after REPL line execution. After that `DataFrame` variable is typed with its own `DataSchema`, so only valid extension properties corresponding to actual columns in DataFrame will be allowed by the compiler and suggested by completion.
```
Comment (Jolanrensen) [inline docs/StardustDocs/topics/extensionPropertiesApi.md:29]: missed two

File: docs/StardustDocs/topics/gradle.md [hunk 2]
```diff
@@ -72,7 +72,7 @@ the same package as file containing the annotation.
 import org.jetbrains.kotlinx.dataframe.annotations.ImportDataSchema
 ```
 
-See KDocs for `ImportDataSchema` in IDE
+See KDocs for `@ImportDataSchema` in IDE
```
Comment (Jolanrensen) [inline docs/StardustDocs/topics/gradle.md:75]: I like this extra @ :)

File: docs/StardustDocs/topics/operations.md [hunk 3]
```diff
@@ -3,8 +3,8 @@
 <!---IMPORT org.jetbrains.kotlinx.dataframe.samples.api.Modify-->
 
 Data transformation pipeline usually consists of several modification operations, such as filtering, sorting, grouping, pivoting, adding/removing columns etc. 
-DataFrame API is designed in functional style so that the whole processing pipeline can be represented as a single statement with a sequential chain of operations.
-`DataFrame` object is immutable and all operations return a new [`DataFrame`](DataFrame.md) instance reusing underlying data structures as much as possible.
+Data frame API is designed in functional style so that the whole processing pipeline can be represented as a single statement with a sequential chain of operations.
```
Comment (Jolanrensen) [inline docs/StardustDocs/topics/operations.md:6]: I think this first is supposed to mean the library, so "The Kotlin DataFrame API", or "The Kotlin DataFrame library API"

File: docs/StardustDocs/topics/overview.md [hunk 4]
```diff
@@ -11,25 +11,31 @@ This documentation is written in such a way that it could be read sequentially a
 
 Data frame is an abstraction for working with structured data. Essentially it’s a 2-dimensional table with labeled columns of potentially different types. You can think of it like a spreadsheet or SQL table, or a dictionary of series objects.
 
-The handiness of this abstraction is not in the table itself but in a set of operations defined on it. `Kotlin Dataframe` library is an idiomatic Kotlin DSL defining such operations. The process of working with data frame is often called *data wrangling* which is the process of transforming and mapping data from one "raw" data form into another format that is more appropriate for analytics and visualization. The goal of data wrangling is to assure quality and useful data.
+The handiness of this abstraction is not in the table itself but in a set of operations defined on it. 
+Kotlin Dataframe Library is an idiomatic Kotlin DSL defining such operations.
```
Comment (Jolanrensen) [inline docs/StardustDocs/topics/overview.md:15]: Library is not with a capital

File: docs/StardustDocs/topics/overview.md [hunk 5]
```diff
@@ -11,25 +11,31 @@ This documentation is written in such a way that it could be read sequentially a
...
+Kotlin Dataframe Library is an idiomatic Kotlin DSL defining such operations. 
+The process of working with data frame is often called *data wrangling* which 
+is the process of transforming and mapping data from one "raw" data form into another format 
+that is more appropriate for analytics and visualization. 
+The goal of data wrangling is to assure quality and useful data.
 
 ## Main Features and Concepts
 
-* [**Hierarchical**](hierarchical.md) — [`DataFrame`](DataFrame.md)  is able to read and present data from different sources including not only plain **CSV** but also **JSON**. That’s why it has been designed hierarchical and allows nesting of columns and cells.
+* [**Hierarchical**](hierarchical.md) — Kotlin DataFrame Library provides an ability to read and present data from different sources including not only plain **CSV** but also **JSON**. That’s why it has been designed hierarchical and allows nesting of columns and cells.
```
Comment (Jolanrensen) [inline docs/StardustDocs/topics/overview.md:23]: L -> l

Comment (Jolanrensen) [reply docs/StardustDocs/topics/overview.md:23]: everywhere in the file :)
"The Kotlin DataFrame library"

**Review by Jolanrensen — COMMENTED** [review]
I like the "GroupBy DataFrame" :) Makes it clearer that it is a form of data frame, but not entirely

File: docs/StardustDocs/topics/overview.md (hunk 5 above)
Comment (zaleslaw) [reply docs/StardustDocs/topics/overview.md:23]: Really? I supposed that L just makes it Cooler!

Comment (Jolanrensen) [reply docs/StardustDocs/topics/overview.md:23]: haha, well unless we put library in the actual name ;P But we just call it "Kotlin DataFrame"

Comment (zaleslaw) [reply docs/StardustDocs/topics/overview.md:23]: Kotlin DataFrame THE Library

**Pull request merged**